### PR TITLE
newsboat: 2.32 -> 2.33

### PR DESCRIPTION
**Pull request by afh**
remove with lib

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/networking/feedreaders/newsboat/default.nix [hunk 1]
```diff
@@ -59,12 +59,12 @@ rustPlatform.buildRustPackage rec {
...
     homepage    = "https://newsboat.org/";
     changelog   = "https://github.com/newsboat/newsboat/blob/${src.rev}/CHANGELOG.md";
     description = "A fork of Newsbeuter, an RSS/Atom feed reader for the text console";
-    maintainers = with maintainers; [ dotlambda nicknovitski ];
-    license     = licenses.mit;
-    platforms   = platforms.unix;
+    maintainers = with lib.maintainers; [ dotlambda nicknovitski ];
+    license     = lib.licenses.mit;
+    platforms   = lib.platforms.unix;
   };
```
Comment (Janik-Haag) [inline pkgs/applications/networking/feedreaders/newsboat/default.nix:69]: why are you removing `with lib;` here? In this scope it's fine we just don't want it in the entire package scope so at start of the file

**Review by Janik-Haag — APPROVED** [review]
Package in general works fine

File: pkgs/applications/networking/feedreaders/newsboat/default.nix (hunk 1 above)
Comment (afh) [reply pkgs/applications/networking/feedreaders/newsboat/default.nix:69]: Thanks for the feedback, @Janik-Haag, much appreciated. I see your point, yet I've seeing `with lib` being removed from the meta section in other packages, e.g. gst123, openxcom. Might be an overzealous misunderstanding from my end.
Are there any explicit nixpkgs guidelines regarding `with lib`, etc. that you could point me to?

**Comment by afh** [comment]
Thanks for merging, @Janik-Haag, very much appreciated 👍
Happy to be pointed to resources for best practices in regards to "improving" packages to comply with them in future PRs.

**Review by nicknovitski — APPROVED** [review]

File: pkgs/applications/networking/feedreaders/newsboat/default.nix (hunk 1 above)
Comment (Janik-Haag) [reply pkgs/applications/networking/feedreaders/newsboat/default.nix:69]: > Happy to be pointed to resources for best practices in regards to "improving" packages to comply with them in future PRs.

From what I know we don't really have formal guide lines for the most part except for [nixpkgs-manual](https://nixos.org/manual/nixpkgs/unstable/) and https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md which I guess you bot already know. The other things I know is basically some more experienced contributor told me at some point or I asked in the nixpkgs dev matrix channel and we came to some conclusion after discussing the topic.

Other then that I think it's fine to do it like this here but I would also be fine with having it for the entire meta scope.

**Review by thiagokokada — APPROVED** [review]

**Pull request merged**